### PR TITLE
Migrated all buttons to buttonv2 and modal to dialog modal in fileupload component

### LINKED ISSUE #4178: Change all buttons to buttonV2 and mui modal to custom modal in File Upload component
**Describe the bug**
Change all buttons to buttonV2 and mui modal to custom modal in File Upload component

Related to updates of https://github.com/coronasafe/care_fe/pull/3716#pullrequestreview-1201945189

### PR DESCRIPTION
**Pull request by patelaryan7751**
## Proposed Changes

- Fixes #4178 
Migrated all buttons to buttonv2 and modal to dialog modal in fileupload component an enhnacement follow up for https://github.com/coronasafe/care_fe/pull/3716#pullrequestreview-1201945189

New Designs-:
## buttons changed to button v2 component
image
image

## modal to dialog modal
image
image
image

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Patient/FileUpload.tsx [hunk 1]
```diff
@@ -564,7 +566,7 @@ export const FileUpload = (props: FileUploadProps) => {
                           <a
                             href={url[item.id]}
                             download={item.name}
-                            className="btn btn-primary m-1 sm:w-auto w-full hover:text-white focus:bg-primary-500"
+                            className="Button outline-offset-1 button-size-default button-shape-square button-primary-default m-1 sm:w-auto w-full hover:text-white focus:bg-primary-500"
                           >
                             <i className="fa-solid fa-circle-arrow-down mr-2"></i>{" "}
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]: Use CareIcons

File: src/Components/Patient/FileUpload.tsx [hunk 2]
```diff
@@ -583,11 +585,11 @@ export const FileUpload = (props: FileUploadProps) => {
                                   setEditFileName(item?.name);
                                   setModalOpenForEdit(true);
                                 }}
-                                className="btn btn-primary m-1 sm:w-auto w-full"
+                                className="m-1 sm:w-auto w-full"
                               >
                                 <i className="fa-solid fa-pencil mr-2"></i>EDIT
                                 FILE NAME
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]: ButtonV2 already applies `flex gap-2`, so `mr-2` will add unnecessary more space.
Also use `CareIcon`
```suggestion
                                <i className="fa-solid fa-pencil" />
                                <span>EDIT FILE NAME</span>
```

File: src/Components/Patient/FileUpload.tsx [hunk 3]
```diff
@@ -658,31 +660,32 @@ export const FileUpload = (props: FileUploadProps) => {
                     </div>
                   </div>
                   <div className="flex flex-wrap items-center">
-                    <label
+                    <ButtonV2
+                      variant="primary"
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]: Default is primary
```suggestion
```

File: src/Components/Patient/FileUpload.tsx [hunk 4]
```diff
@@ -1115,183 +1121,184 @@ export const FileUpload = (props: FileUploadProps) => {
...
+            </ButtonV2>
+            <ButtonV2
+              variant="secondary"
+              type="button"
+              className="mr-2 w-full md:w-auto"
+              disabled={btnloader}
+              onClick={() => setModalOpenForEdit(false)}
+            >
+              Cancel
+            </ButtonV2>
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]: Merge the latest develop and use the generalized `<Cancel onClick={...} /` button.

File: src/Components/Patient/FileUpload.tsx [hunk 5]
```diff
@@ -1115,183 +1121,184 @@ export const FileUpload = (props: FileUploadProps) => {
...
+              disabled={btnloader}
+              className="mr-2 w-full md:w-auto"
+            >
+              <i
+                className={`fa-solid fa-spinner mr-2 animate-spin ${
+                  !btnloader ? " hidden " : " "
+                }`}
+              ></i>
+              Proceed
+            </ButtonV2>
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]: Merge the latest develop and use the generalized `<Submit onClick={...} />` button. This is to make the design consistent across the platform.

For the loader, you can override the children.

File: src/Components/Patient/FileUpload.tsx [hunk 6]
```diff
@@ -1115,183 +1121,184 @@ export const FileUpload = (props: FileUploadProps) => {
...
+              rows={6}
+              multiline
+              required
+              className="w-full border p-2 max-h-64"
+              placeholder="Type the reason..."
+              margin="dense"
+              value={archiveReason}
+              onChange={(e) => setArchiveReason(e.target.value)}
+              errors={archiveReasonError}
+            />
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]: Use the tailwind component: `TextAreaFormField`

File: src/Components/Patient/FileUpload.tsx [hunk 7]
```diff
@@ -1115,183 +1121,184 @@ export const FileUpload = (props: FileUploadProps) => {
...
+          className="flex flex-col w-full my-4 mx-2"
+        >
+          <div>
+            <InputLabel
+              className="text-md text-black text-center"
+              id="archivereasonlabel"
+            >
+              Please state the reason for archiving <b>{modalDetails?.name}</b>{" "}
+              file?
+            </InputLabel>
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]: Use the tailwind component `FieldLabel`

File: src/Components/Patient/FileUpload.tsx [hunk 8]
```diff
@@ -1115,183 +1121,184 @@ export const FileUpload = (props: FileUploadProps) => {
...
-              </button>
+            <div className="text-md text-center">
+              <b>Time of Archive:</b>
+              {formatDate(modalDetails?.archiveTime)}
             </div>
           </div>
+          <div className="flex flex-col-reverse md:flex-row gap-2 mt-4 justify-end">
+            <ButtonV2
+              type="button"
+              variant="danger"
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]: Close button is not under danger context. Use secondary 
```suggestion
              variant="secondary"
```

**Comment by patelaryan7751** [comment]
Hey @rithviknishad i have done the changes specific to this issue only. Furthermore there is more to migrate which can be dealt with seperate issue.

File: src/Components/Patient/FileUpload.tsx [hunk 9]
```diff
@@ -1144,53 +1149,31 @@ export const FileUpload = (props: FileUploadProps) => {
...
-            <InputLabel
+            <FieldLabel
               className="text-md text-gray-500"
               id="editfilenamelabel"
             >
               Please enter the file name
-            </InputLabel>
-            <TextInputField
+            </FieldLabel>
+            <TextFormField
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx:1158]:
```suggestion
            <TextFormField
              label="File name"
```

File: src/Components/Patient/FileUpload.tsx [hunk 10]
```diff
@@ -1144,53 +1149,31 @@ export const FileUpload = (props: FileUploadProps) => {
           className="flex flex-col w-full"
         >
           <div>
-            <InputLabel
+            <FieldLabel
               className="text-md text-gray-500"
               id="editfilenamelabel"
             >
               Please enter the file name
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]:
```suggestion
```

File: src/Components/Patient/FileUpload.tsx [hunk 11]
```diff
@@ -1144,53 +1149,31 @@ export const FileUpload = (props: FileUploadProps) => {
...
-            <ButtonV2
-              variant="secondary"
-              type="button"
-              className="mr-2 w-full md:w-auto"
-              disabled={btnloader}
-              onClick={() => setModalOpenForEdit(false)}
-            >
-              Cancel
-            </ButtonV2>
+            <Cancel type="button" onClick={() => setModalOpenForEdit(false)} />
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]:
```suggestion
            <Cancel onClick={() => setModalOpenForEdit(false)} />
```

File: src/Components/Patient/FileUpload.tsx [hunk 12]
```diff
@@ -1144,53 +1149,31 @@ export const FileUpload = (props: FileUploadProps) => {
...
+            <Submit disabled={btnloader} label={"Proceed"} />
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]:
```suggestion
            <Submit disabled={btnloader} label="Proceed" />
```

File: src/Components/Patient/FileUpload.tsx [hunk 13]
```diff
@@ -1115,183 +1126,136 @@ export const FileUpload = (props: FileUploadProps) => {
...
+          </div>
+        </form>
+      </DialogModal>
+      <DialogModal
+        show={modalOpenForArchive}
+        title={
+          <div className="flex flex-row">
+            <div className="text-center my-1 mr-3 rounded-full bg-red-100 py-4 px-5">
+              <CareIcon className="care-l-exclamation-triangle text-lg text-red-500 " />
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]:
```suggestion
              <CareIcon className="care-l-exclamation-triangle text-lg text-danger-500" />
```

File: src/Components/Patient/FileUpload.tsx [hunk 14]
```diff
@@ -1115,183 +1126,136 @@ export const FileUpload = (props: FileUploadProps) => {
...
+          className="flex flex-col w-full"
+        >
+          <div>
+            <FieldLabel
+              className="text-md text-gray-500"
+              id="editfilenamelabel"
+            >
+              Please enter the file name
+            </FieldLabel>
+            <TextFormField
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]:
```suggestion
            <TextFormField
              label="Enter file name"
```

File: src/Components/Patient/FileUpload.tsx [hunk 15]
```diff
@@ -1115,183 +1126,136 @@ export const FileUpload = (props: FileUploadProps) => {
...
+            archiveFile(modalDetails?.id, archiveReason);
+          }}
+          className="flex flex-col w-full my-4 mx-2"
+        >
+          <div>
+            <FieldLabel className="text-md text-black" id="archivereasonlabel">
+              Please state the reason for archiving <b>{modalDetails?.name}</b>{" "}
+              file?
+            </FieldLabel>
+            <TextAreaFormField
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]:
```suggestion
            <TextAreaFormField
              label={<span>State the reason for archiving <b>{modalDetails?.name}</b> file?</span>}
```

File: src/Components/Patient/FileUpload.tsx [hunk 16]
```diff
@@ -1115,183 +1126,136 @@ export const FileUpload = (props: FileUploadProps) => {
...
+              value={archiveReason}
+              onChange={(e) => setArchiveReason(e.value)}
+              error={archiveReasonError}
+            />
+          </div>
+          <div className="flex flex-col-reverse md:flex-row gap-2 mt-4 justify-end">
+            <Cancel
+              type="button"
+              onClick={() => setModalOpenForArchive(false)}
+            />
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]:
```suggestion
            <Cancel onClick={() => setModalOpenForArchive(false)} />
```

File: src/Components/Patient/FileUpload.tsx [hunk 17]
```diff
@@ -1115,183 +1126,136 @@ export const FileUpload = (props: FileUploadProps) => {
...
+            <div className="text-md text-center">
+              <b>Time of Archive:</b>
+              {formatDate(modalDetails?.archiveTime)}
             </div>
           </div>
+          <div className="flex flex-col-reverse md:flex-row gap-2 mt-4 justify-end">
+            <Cancel
+              type="button"
+              onClick={(_) => setModalOpenForMoreDetails(false)}
+            />
```
Comment (rithviknishad) [inline src/Components/Patient/FileUpload.tsx]:
```suggestion
            <Cancel onClick={() => setModalOpenForMoreDetails(false)} />
```

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**